---
title: "TFM2_18S_amplicons_analysis"
author: "lrubinat"
date: "3/9/2017"
output:
  html_document:
    theme: united
    toc: yes
  pdf_document:
    highlight: zenburn
    toc: yes
---


<!--- INITIALIZATION
```{r, echo=FALSE}
#error hook to kill knitr in case of errors
library(knitr)
knit_hooks$set(error = function(x, options) stop(x))
opts_chunk$set(cache=TRUE, autodep=TRUE)
```
--->

```{r libraries, echo=T, message=F}
library(data.table)
library(dplyr)
library(DT)
library(ggplot2)
library(gtools)
library(vegan)
library(iNEXT)
library(fossil)
```


# 1) 18S amplicons

## 1.1) Data overview

Let's read the dataset and remove the samples containing less than 49975 reads:

``` {r load_data, echo=FALSE, message=FALSE}
setwd("~/Documents/TFM2/TFM2_18S_amplicons_analysis/")

#read data 
tb18_tax <- read.table(file="~/Documents/TFM_returns/TFM/home/data/MALASPINA/Malaspina_18S_Surface/OTUtb_18S_curated_46808_OTUs.txt", head=TRUE, fill=TRUE)

#table dimensions and format before setting column names
dim(tb18_tax) # 46808   132
tb18_tax[1:5,1:5]

row.names(tb18_tax)<-tb18_tax[,1]
tb18_tax<-tb18_tax[,-1]
tb18_tax[is.na(tb18_tax)]<-"NA"

dim(tb18_tax) #46808   131
tb18_tax[1:5,1:5]

#drop repeated samples in stations 68 and 71:
tb18_tax<-subset(tb18_tax, select=-c(st068a_MD1254,st071a_MD1318))
dim(tb18_tax)

#table with taxonomi classification alone
tb18_class <- tb18_tax[,123:129]
dim(tb18_class) #46808   7
tb18_class[1:5,1:7]

#table with occurence data alone
tb18_tax_occur <- tb18_tax[,1:122]
dim(tb18_tax_occur) #46808   122
tb18_tax_occur[1:5,1:5]

amplicons_per_sample_tb18<-colSums(tb18_tax_occur)
amplicons_per_sample_tb18[which(colSums(tb18_tax_occur)<49975)]
#initially we rarefied at 49975 reads; in this new version analysis
#we remove all samples with less than 49975 reads.

#remove samples with less than 49975 reads
tb18_tax_occur_min49975 <- tb18_tax_occur[,colSums(tb18_tax_occur) >= 49975]
dim(tb18_tax_occur_min49975) #46808   105

#remove samples with omitted in 16S dataset (so that we can compare the relative abundance of 16S and 18S OTUs considering the same samples)
#tb18_tax_occur_min49975<-subset(tb18_tax_occur_min49975, select=-c(st122, st124, st137, st144))
dim(tb18_tax_occur_min49975) #46808   105

#let's check if we loose any OTU when excluding the mentioned stations.
tb18_tax_occur_min49975_no_cero<-tb18_tax_occur_min49975[,-(which(colSums(tb18_tax_occur_min49975)==0))]
dim(tb18_tax_occur_min49975_no_cero)
```

Table dimensions and content outline:

```{r starting_dataset, echo=FALSE}
dim(tb18_tax_occur_min49975)
tb18_tax_occur_min49975[1:5,1:5]
```

Minimum number of reads per station:

```{r reads_per_sample_overview1, echo=1}
min(colSums(tb18_tax_occur_min49975)) 
#49975
```

Maximum number of reads per station:

```{r reads_per_sample_overview2, echo=1}
max(colSums(tb18_tax_occur_min49975)) 
# max: 935755
```

Identification of stations with higher number of reads:

```{r reads_per_sample_overview3, echo=TRUE}
amplicons_per_sample<-colSums(tb18_tax_occur_min49975)
amplicons_per_sample[which(colSums(tb18_tax_occur_min49975)>300000)]
```

Overall reads per sample:

``` {r reads_per_sample_overview4, echo=FALSE}
plot(sort(colSums(tb18_tax_occur_min49975)), pch=19, xlab="sample", ylab="reads per sample", cex=0.9)
```


## 1.2) Normalization

Let's normalize the original dataset by randomly subsampling 49975 reads in each station:

``` {r species_richness_rarefaction1, echo=TRUE}
tb18_tax_occur_min49975_t<-t(tb18_tax_occur_min49975)
tb18_tax_occur_ss49975<-rrarefy(tb18_tax_occur_min49975_t, 49975)
```

The normalized table shows the following dimensions and format:

```{r species_richness_rarefaction2, echo=FALSE}
dim(tb18_tax_occur_ss49975)
tb18_tax_occur_ss49975[1:5,1:5]
```

Its content fits with the expected normalization values (49975 reads per station):

``` {r species_richness_rarefaction3, echo=TRUE}
rowSums(tb18_tax_occur_ss49975)
```

Let's check out how many OTUs don't appear in the new table:

```{r species_richness_rarefaction4, echo=1:5}
length(which(colSums(tb18_tax_occur_ss49975)==0)) #5992
```

There are 20487 OTUs that don't show any occurrence in the normalized data. Let's remove them from the table and take a look at its final dimensions:

```{r species_richness_rarefaction5, echo=1:3}
tb18_tax_occur_ss49975_no_cero<-tb18_tax_occur_ss49975[,-(which(colSums(tb18_tax_occur_ss49975)==0))]
dim(tb18_tax_occur_ss49975_no_cero) #105 40816

#the final dimensions of the normalized table are 117 26657.
#40816 + 5992 = 46808
```

Datasets summary:
dim(tb18_tax) --> 46808   129
dim(tb18_tax_occur) --> 46808   122
dim(tb18_tax_occur_ss49975_no_cero) --> 105 40816




## 1.3) General community analysis

### 1.3.1) Richness and evenness (Shannon index)

```{r shannon_index1, echo=FALSE}
tb18_tax_occur_ss49975_div <- diversity(tb18_tax_occur_ss49975_no_cero, index="shannon")
```

Most of the samples take Shannon Index values between 5.5 and 6:

```{r shannon_index2, echo=FALSE}
boxplot(tb18_tax_occur_ss49975_div, pch=19, main="Shannon's index of diversity")
plot(sort(tb18_tax_occur_ss49975_div), pch=19, main="Shannon's index of diversity")
```

### 1.3.2) Richness: OTU number

```{r richness_otu_no1, echo=FALSE}
OTUs_per_sample_18S_tax_occur_ss49975<-specnumber(tb18_tax_occur_ss49975_no_cero)
```

Lowest number of OTUs per sample:

```{r richness_otu_no2, echo=FALSE}
min(OTUs_per_sample_18S_tax_occur_ss49975)
```

Maximum number of OTUs per sample:

```{r richness_otu_no3, echo=FALSE}
max(OTUs_per_sample_18S_tax_occur_ss49975)
```

In most of the samples, we can identify between 3000 and 4500 OTUs:

```{r richness_otu_no4, echo=TRUE}
plot(sort(OTUs_per_sample_18S_tax_occur_ss49975), pch=19)
boxplot(OTUs_per_sample_18S_tax_occur_ss49975, pch=19)
```

### 1.3.3) Index of evenness

#### 1.3.3.1) Pielou's index

```{r pielou_index_of_evenness1, echo=TRUE}
pielou_evenness_18S_tax_occur_ss49975 <- tb18_tax_occur_ss49975_div/log(OTUs_per_sample_18S_tax_occur_ss49975)
```

The Pielou index (constrained between 0 and 1) takes values closer to 1 as the variation of species proportion in a sample decreases. Most of the samples get values between 0.7 and 0.8, meaning that the numerical composition of different OTUs in a sample is highly similar:

```{r pielou_index_of_evenness2, echo=TRUE}
plot(sort(pielou_evenness_18S_tax_occur_ss49975), pch=19)
boxplot(pielou_evenness_18S_tax_occur_ss49975, pch=19)
```

The OTU_6, with 196592 reads, is the most abundant in the overall dataset:

```{r OTUs_overall_abundance, echo=TRUE}
head(sort(colSums(tb18_tax_occur_ss49975_no_cero), decreasing=T), n=10L)
```

Most of the OTUs show very few occurrences; the plot suggests that we will probably be able to identify a significant ammount of rare otus:

```{r OTUs_overall_abundance2, echo=TRUE}
plot(log(sort(colSums(tb18_tax_occur_ss49975_no_cero), decreasing=T)), pch=19)
```

<!---
#### 1.3.3.2) Sads
library(sads)
?sads
--->

### 1.3.4) Abundance Models
#### 1.3.4.1) Rank-Abundance or Dominance/Diversity Model ("radfit")

The OTUs abundance distribution fits relativelly close to log-normal model. 

```{r radfit, echo=FALSE}
#?radfit
#otu_tb18_t[1:5,1:5]

tb18_tax_occur_min49975_radfit<-radfit(colSums(tb18_tax_occur_min49975_t))
plot(tb18_tax_occur_min49975_radfit)
```

#### 1.3.4.2) Preston's Lognormal Model

According to Preston's lognormal model fit into species frequencies groups, we're missing ~2451 species:

```{r preston_model1, echo=T}
tb18_tax_occur_ss49975_prestonfit<-prestonfit(colSums(tb18_tax_occur_min49975_t))
plot(tb18_tax_occur_ss49975_prestonfit, main="Pooled species")

veiledspec(tb18_tax_occur_ss49975_prestonfit)
```

When computing Prestons' lognormal model fit without pooling data into groups, we seem to miss ~2319 species:

```{r preston_model2, echo=4}
tb18_tax_occur_ss49975_dist_all<-prestondistr(colSums(tb18_tax_occur_min49975_t))
plot(tb18_tax_occur_ss49975_prestonfit, main="All malaspina")
lines(tb18_tax_occur_ss49975_dist_all, line.col="blue3")

veiledspec(tb18_tax_occur_ss49975_dist_all)
```


### 1.3.5) Rarefaction curve
<!---
(To be computed)
--->

```{r rarefraction_curve, echo=TRUE}
?rarecurve
rarecurve(t(colSums(tb18_tax_occur_ss49975_no_cero)))


test<-colSums(tb18_tax_occur_ss49975_no_cero)
rarecurve(test, step = 1, 50000, xlab = "Sample Size", ylab = "OTUs", label = TRUE)
#rarecurve(colSums(tb18_tax_occur_min49975_no_cero), step = 1, 50000, xlab = "Sample Size", ylab = "OTUs", label = TRUE)

str(colSums(tb18_tax_occur_ss49975_no_cero))

tb18_tax_occur_min49975_t_colSums<-colSums(tb18_tax_occur_min49975_t)
str(tb18_tax_occur_min49975_t)

tb18_tax_occur_min49975_t_colSums<-as.matrix(tb18_tax_occur_min49975_t_colSums)

tb18_tax_occur_min49975_t_colSums<-t(tb18_tax_occur_min49975_t_colSums)

tb18_tax_occur_min49975_t_colSums[,1:3]


rarecurve(tb18_tax_occur_min49975_t_colSums, step = 1, 50000, xlab = "Sample Size", ylab = "OTUs", label = TRUE)



rarec_input<-t(as.matrix(colSums(tb18_tax_occur_ss49975_no_cero)))
tb18_rarecurve50000<-rarecurve(rarec_input, step = 1, 50000, xlab = "Sample Size", ylab = "OTUs", label = TRUE)
tb18_rarecurve50000
head(tb18_rarecurve50000)

tb18_rarecurve100000<-rarecurve(rarec_input, step = 1, 100000, xlab = "Sample Size", ylab = "OTUs", label = TRUE)
tb18_rarecurve100000
head(tb18_rarecurve100000)
#rarecurve(rarec_input, step = 1, 100000, xlab = "Sample Size", ylab = "OTUs", label = TRUE)

rarec_input.raw<-t(as.matrix(colSums(tb18_tax_occur)))
tb18_rarecurve100000.raw<-rarecurve(rarec_input.raw, step = 1, 100000, xlab = "Sample Size", ylab = "OTUs", label = TRUE)
tb18_rarecurve100000.raw
head(tb18_rarecurve100000.raw)
#rarecurve(rarec_input, step = 1, 100000, xlab = "Sample Size", ylab = "OTUs", label = TRUE)
```


### 1.3.6) Beta diversity

#### 1.3.6.1) Dissimilarity matrix using Bray-Curtis index:

The Bray-Curtis dissimilarity, constrained between 0 (minimum distance) and 1 (highest dissimilarity) allows us to quantify the differences between samples according to the composition and relative abundance of their OTUs. In our dataset, most of the samples pairs take dissimilarity values between between 6.5 and 7.5, meaning that their composition is substantially different.

```{r beta_div1, echo=FALSE}
#?vegdist
tb18_tax_occur_ss49975_no_cero.bray<-vegdist(tb18_tax_occur_ss49975_no_cero, method="bray")
boxplot(tb18_tax_occur_ss49975_no_cero.bray, main="Bray-Curtis dissimilarity matrix")
```

#### 1.3.6.2) Hierarchical clustering

The stations seem to form clusters according to geographic localization, but there are no evident clusters separated from the general groups.

(To be done: assign Longhurst provinces information to each station and check if any of the central clusters is meaningful regarding to the samples' geographical location)

```{r beta_div2, echo=FALSE}
#UPGMA
tb18_tax_occur_ss49975_no_cero.upgma<-hclust(tb18_tax_occur_ss49975_no_cero.bray, "average")
plot(tb18_tax_occur_ss49975_no_cero.upgma, cex=.35, main="Samples Hierarchical Clustering")
```

#### 1.3.6.3) Non-metric multidimensional scaling

We can identify a prominent group in the central part of the NMDS plot and a few outliers in the central-high edge of the plot. The stress parameter takes a value below 0.2, suggesting that the plot is acceptable. 

```{r monoNMDS, echo=F}
#NMDS
tb18_tax_occur_ss49975_no_cero.nmds<-monoMDS(tb18_tax_occur_ss49975_no_cero.bray)
tb18_tax_occur_ss49975_no_cero.nmds
plot(tb18_tax_occur_ss49975_no_cero.nmds, main="monoMDs method")
```

When implementing a most robut function for computing NMDS plots, the result is quiet the same:

```{r metaNMDS, echo=F}
tb18_tax_occur_ss49975_no_cero.meta_nmds<-metaMDS(tb18_tax_occur_ss49975_no_cero.bray)
plot(tb18_tax_occur_ss49975_no_cero.meta_nmds, main="metaMDS method")
```

## 1.4) Geographical analysis

```{r load_geo_data, echo=F, results="hide", message=F}
#load geographical ubication of stations and sort according to otu_tb18 stations sequence.
MP_geo_18S<-read.table(file="~/Documents/TFM2/data/MALASPINA/geo/mp_surface_ubication_modif.txt", sep="\t", header=T)

MP_geo_18S<-data.table(MP_geo_18S)
MP_geo_18S[,st_name:=(paste("st",MP_geo_18S$station,"_",MP_geo_18S$code,sep=""))]
row.names(MP_geo_18S)<-MP_geo_18S$st_name

dim(MP_geo_18S)
MP_geo_18S[1:5,1:5]
tb18_tax_occur_ss49975_no_cero[1:5,1:5]

#read lat-long in decimal degrees and translate into distance in km.
#library(fossil)

#select only columns containing info about station, latitude and longitude.
MP_geo_18S_reduced<-create.lats(MP_geo_18S, loc="OTUId", long="long", lat="lat")
head(MP_geo_18S_sorted)

#create a distance matrix (lower triangle) between a list of points.
geo_distances_MP_18S<-earth.dist(MP_geo_18S_sorted_v2, dist = TRUE)
head(geo_distances_MP_18S)
dim(geo_distances_MP_18S)

geo_distances_MP_18S<-as.matrix(geo_distances_MP_18S)
dim(geo_distances_MP_18S)

#geo distances dataset ready to use "geo_distances_MP_18S"
```

Working datasets:

1) Community matrix: tb18_tax_occur_ss49975_no_cero

```{r working_datasets1, echo=T}
dim(tb18_tax_occur_ss49975_no_cero)
tb18_tax_occur_ss49975_no_cero[1:5, 1:5]
```

2) Community Bray-Curtis: tb18_tax_occur_ss49975_no_cero.bray

```{r working_datasets2, echo=2}
tb18_tax_occur_ss49975_no_cero.bray<-as.matrix(tb18_tax_occur_ss49975_no_cero.bray)
dim(tb18_tax_occur_ss49975_no_cero.bray)
```

3) Stations distances in km: geo_distances_MP_18S

```{r working_datasets3, echo=T}
dim(geo_distances_MP_18S)
```

Communities quickly change their composition across geographical distances:

```{r working_datasets4, echo=T}
plot(geo_distances_MP_18S, tb18_tax_occur_ss49975_no_cero.bray, pch=19, cex=0.4, xlab="Geopgraphical distances", ylab="Bray-Curtis dissimilarities")
```

### 1.4.1) Mantel correlograms

Mantel statistic is -significantlly- so low, meaning that the correlation between samples dissimilarity and geographical distances is weak.

```{r mantel_correlogram1, echo=T}
mantel(geo_distances_MP_18S, tb18_tax_occur_ss49975_no_cero.bray)
```

Maximum distance between samples:

```{r mantel_correlogram2, echo=F}
max(geo_distances_MP_18S)
```

Minimum distance between samples:

```{r mantel_correlogram3, echo=F}
min(geo_distances_MP_18S)
```

Correlograms:

```{r mantel_correlogram4, echo=T}
MP_18s_ss49975_mantel_correl_by_1000km<-mantel.correlog(tb18_tax_occur_ss49975_no_cero.bray, D.geo=geo_distances_MP_18S, break.pts=seq(0,20000, by=1000))
plot(MP_18s_ss49975_mantel_correl_by_1000km)

MP_18s_ss49975_mantel_correl_by_100km<-mantel.correlog(tb18_tax_occur_ss49975_no_cero.bray, D.geo=geo_distances_MP_18S, break.pts=seq(0,20000, by=100))
plot(MP_18s_ss49975_mantel_correl_by_100km)
```

## 1.5) Abundance vs. occurence

```{r OTUs_mean_relative_abund, echo=F, results="hide"}
tb18_tax_occur_ss49975_no_cero[1:5,1:5]
tb18_tax_occur_ss49975_no_cero_t<-t(tb18_tax_occur_ss49975_no_cero)

colSums(tb18_tax_occur_ss49975_no_cero_t)

#local abundance percentage
tb18_tax_occur_ss49975_no_cero_t.rabund<-tb18_tax_occur_ss49975_no_cero_t/49975

colSums(tb18_tax_occur_ss49975_no_cero_t.rabund)
tb18_tax_occur_ss49975_no_cero_t.rabund[1:5,1:5]

#OTUs mean relative abundance
tb18_tax_occur_ss49975_no_cero_t.rabund_means<-rowMeans(tb18_tax_occur_ss49975_no_cero_t.rabund) 
tb18_tax_occur_ss49975_no_cero_t.rabund_means<-as.data.frame(tb18_tax_occur_ss49975_no_cero_t.rabund_means)

head(tb18_tax_occur_ss49975_no_cero_t.rabund_means)
```

```{r OTUs_occurence, echo=F, results='hide'}
tb18_tax_occur_ss49975_no_cero_t.rabund.occur<-tb18_tax_occur_ss49975_no_cero_t.rabund
tb18_tax_occur_ss49975_no_cero_t.rabund.occur[tb18_tax_occur_ss49975_no_cero_t.rabund.occur>0]<-1
tb18_tax_occur_ss49975_no_cero_t.rabund.occur[1:5,1:5] ### presence - absence table

#percentage of occurence in overall stations
tb18_tax_occur_ss49975_no_cero_t.rabund_means.occurence_perc<-as.data.frame(100*(rowSums(tb18_tax_occur_ss49975_no_cero_t.rabund.occur)/117))

str(tb18_tax_occur_ss49975_no_cero_t.rabund_means.occurence_perc)
```

```{r merge_rabund_peroccur, echo=F, results='hide'}
otu_tb18_ss49975_rabund_percoccur<-merge(tb18_tax_occur_ss49975_no_cero_t.rabund_means,tb18_tax_occur_ss49975_no_cero_t.rabund_means.occurence_perc, by="row.names")

colnames(otu_tb18_ss49975_rabund_percoccur)<-c("OTUs","mean_rabund","perc_occur")
otu_tb18_ss49975_rabund_percoccur[1:5,]

row.names(otu_tb18_ss49975_rabund_percoccur)<-otu_tb18_ss49975_rabund_percoccur[,1]
otu_tb18_ss49975_rabund_percoccur<-otu_tb18_ss49975_rabund_percoccur[,-1]
otu_tb18_ss49975_rabund_percoccur[1:5,]
```

In the following plot, we can appreciate the OTUs distribution according to their percentage of occurence and relative abundance. The red line keeps up OTUs that occur in more than 80% of the samples, the green line limits regionally rare OTUs (< 0.001%), and the blue one restricts regionally abundant OTUs (> 0.1%).

```{r abund_vs_occurence_table, echo=F}
plot(otu_tb18_ss49975_rabund_percoccur$mean_rabund,otu_tb18_ss49975_rabund_percoccur$perc_occur, log="x", pch=19, cex=0.8, xlab="Mean relative abundance", ylab="Percentage of occurence")
abline(h=80, col="red") #occurence higher than 80%
abline(v=0.00001, col="green") #rare OTUs
abline(v=0.001, col="blue") #cosmopolitan OTUs

#Conventional limits:
#Regionally rare     = 0.00001
#Regionally abundant = 0.001
```

Regionally abundant OTUs (relative abundance over 0.1%):

```{r abundant_OTUs, echo=7}
#regionally abundant
tb18_ss49975_abundant<-otu_tb18_ss49975_rabund_percoccur[otu_tb18_ss49975_rabund_percoccur$mean_rabund > 0.001,]

tb18_ss49975_abundant_sorted<-tb18_ss49975_abundant[order(tb18_ss49975_abundant$mean_rabund, tb18_ss49975_abundant$perc_occur, decreasing = T), c(1,2)]

tb18_ss49975_abundant_sorted
dim(tb18_ss49975_abundant_sorted)
```

Proportion of regionally abundant OTUs (%):

```{r abundant_OTUs2, echo=F}
#there are 83 regionally abundant OTUs.
(162/25191)*100 # = 7.29% of the OTUs are regionally abundant

#length(row.names(otu_tb18_ss49975_rabund_percoccur[otu_tb18_ss49975_rabund_percoccur$mean_rabund > 0.001,])) # 83 OTUs
#row.names(otu_tb18_ss49975_rabund_percoccur[otu_tb18_ss49975_rabund_percoccur$mean_rabund > 0.001,])
```

Cosmopolitan OTUs (relative abundance over 0.1% and occurence in more than 80% of samples):

```{r select_cosmopolitan, echo=6}
otu_tb18_ss49975_rabund_cosm<-otu_tb18_ss49975_rabund_percoccur[otu_tb18_ss49975_rabund_percoccur$mean_rabund > 0.001,]
otu_tb18_ss49975_rabund_poccur_cosm<-otu_tb18_ss49975_rabund_cosm[otu_tb18_ss49975_rabund_cosm$perc_occur > 80,]
otu_tb18_ss49975_cosmop_sorted<-otu_tb18_ss49975_rabund_poccur_cosm[order(otu_tb18_ss49975_rabund_poccur_cosm$perc_occur, otu_tb18_ss49975_rabund_poccur_cosm$mean_rabund, decreasing = T), c(1,2)]

otu_tb18_ss49975_cosmop_sorted
dim(otu_tb18_ss49975_cosmop_sorted)
```

Proportion of cosmopolitan OTUs (%):

```{r percentage_cosmopolitan, echo=F}
(64/25191)*100
```

Number and proportion (%) of rare OTUs:

```{r rare_OTUs, echo=1}
dim(otu_tb18_ss49975_rabund_percoccur[otu_tb18_ss49975_rabund_percoccur$mean_rabund < 0.00001 & otu_tb18_ss49975_rabund_percoccur$mean_rabund >0,])
 
(18551/25191)*100 # = 28.56% of the OTUs are regionally rare
```

<!---
```{r otu_col_chech, echo = T}
dim(otu_tb18_ss49975_rabund_percoccur)
dim(tb18_tax_occur_min49975)

#rare OTUs:
# length(row.names(otu_tb16_ss49975_rabund_percoccur[otu_tb16_ss49975_rabund_percoccur$mean_rabund < 0.00001 & otu_tb16_ss49975_rabund_percoccur$mean_rabund >0 ,])) # 325 OTUs
```
--->

```{r iNEXT_test, echo=F}

iNEXT(spider, q=0, datatype="abundance")
tb18S_inext<-iNEXT(colSums(tb18_tax_occur_ss49975_no_cero), q=0, datatype = "incidence_raw")
data(spider)
str(spider)

tb18_tax_occur_ss49975_no_cero_t_binary<-ifelse(t(tb18_tax_occur_ss49975_no_cero)>0, 1, 0)
test[1:5,1:5]
tb18S_inext<-iNEXT(t(tb18_tax_occur_ss49975_no_cero_t_binary), q=0, datatype ="incidence_raw")

tb18S_inext<-iNEXT(t(tb18_tax_occur_ss49975_no_cero), q=0, datatype ="incidence_freq")

tb18S_inext<-iNEXT(t(tb18_tax_occur_ss49975_no_cero), q=0, datatype ="abundance")
dim(tb18_tax_occur_ss49975_no_cero_t_binary)
```


```{r iNEXT_list, echo=F}

rarecurve_comparison_list<-vector("list")
rarecurve_comparison_list$no_rarefied<-t(tb18_tax_occur)
rarecurve_comparison_list$rarefied_ss49975<-tb18_tax_occur_ss49975_no_cero

#parks.spp <- vector("list") 
#parks.spp$park1 <- park1.spp.abund
#parks.spp$park2 <- park2.spp.abund

rarecurve(t(as.matrix(tb18_tax_occur_ss49975_no_cero)))
tb18_tax_occur_ss49975_no_cero.rarecurve<-rarecurve(t(as.matrix(tb18_tax_occur_ss49975_no_cero)))
head(tb18_tax_occur_ss49975_no_cero.rarecurve)

#rarecurve(t(as.matrix(park1.spp.abund)))
#park1.rarecurve <- rarecurve(t(as.matrix(park1.spp.abund)))
#head(park1.rarecurve)

# iNEXT: individual-based abundance data (comparing sites)
tb18_rarecurve_comparison<-iNEXT(rarecurve_comparison_list, q=0, datatype="abundance")
ggiNEXT(tb18_rarecurve_comparison)
```


